querybuilder.go

```diff
@@ -89,7 +89,7 @@ func (qb QueryBuilder) Filter(qo queryoptions.Options) (bson.M, error) {
 			}
 
 			switch bsonType {
-			case "array":
+			case "array", "object", "string":
```

Comment:
Such a great cleanup here... I like how you consolidate the cases (both here and for date types and for numeric types below)